### ISSUE TITLE
[NOT ISSUE]   HELP NEED for ECIES implementation  

### ISSUE THREAD

**Issue by duanbing** [briansmith/ring#979]
The question link is here:  https://crypto.stackexchange.com/questions/80468/any-one-who-is-familiar-with-ring-crypto-repo-can-help-to-review-my-code-of-ecie

Could anyone help take a look at this issue? 
@briansmith

**Comment by briansmith** [comment]
@duanbing if you need ECIES support in *ring* I'm happy to help you add it if you submit a PR or even a sketch of how you want to add it to *ring*. But, it's not reasonable for me to spend the time to dig through a fork of *ring* and try to understand everything that might have changed there.

**Comment by duanbing** [comment]
@briansmith  Thanks, you're right, PR is what I want to do finally, but before I submit a PR, I need to make it work. 😝.  
As the link shows, I use private_key_ops.point_mul to compute the shared secret. but the problem is each time I calculate the  $$ r \times K_b \ne  k_B \times R$$.  

And the sketch is presented in the link [above](https://crypto.stackexchange.com/questions/80468/any-one-who-is-familiar-with-ring-crypto-repo-can-help-to-review-my-code-of-ecie).   and the code is here: [ecies](https://github.com/duanbing/xchain-rust-crypto/blob/master/src/ec/suite_b/ecies.rs#L277). 

I did some changes there:
1.  I changed the way to generate the ECDSA private key by providing the seed outside. but I think it's right because the `sign` and `verify` work well;  
2.  almost replace all the place where needs a random generator with rand::thread_rng().fill(&mut buf).
3.  expose some new methods about fetching curve point in `ring/signature` and EcdsaKeyPair for other purposes.  

 Since those changes are needed by my work, I don't think you will accept it as a PR.  **But ECIES work can work well based on both this and the master code in my plan**, and most of the changes are very careful.  Thanks for your previous work.  

If more info should be provided here, plz let me know, thanks for your kind reply. 
@briansmith
